exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_exporter.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "nlohmann/json.hpp"
+#include "opentelemetry/ext/http/client/curl/http_client_curl.h"
+#include "opentelemetry/nostd/type_traits.h"
+#include "opentelemetry/sdk/logs/exporter.h"
+#include "opentelemetry/sdk/logs/log_record.h"
+
+#include <time.h>
+#include <iostream>
+
+namespace nostd   = opentelemetry::nostd;
+namespace sdklogs = opentelemetry::sdk::logs;
+using json        = nlohmann::json;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * Struct to hold Elasticsearch exporter configuration options.
+ */
+struct ElasticsearchExporterOptions
+{
+  // Configuration options to establish Elasticsearch connection
+  std::string host_;
+  int port_;
+  std::string index_;
+
+  // Maximum time to wait for response after sending request to Elasticsearch
+  int response_timeout_;
+
+  // Whether to print the status of the exporter in the console
+  bool console_debug_;
+
+  /**
+   * Constructor for the ElasticsearchExporterOptions. By default, the endpoint is
+   * localhost:9200/logs with a timeout of 30 seconds and disabled console debugging
+   * @param host The host of the Elasticsearch instance
+   * @param port The port of the Elasticsearch instance
+   * @param index The index/shard that the logs will be written to
```

Comment:
just wondering - is this `index` variable being used anywhere? i think in the reqest body, you just have default `index:{}` right?

Reply:
the way elasticsearch works is you connect to it with an endpoint, like `localhost::3000/logs` for example. The `/logs` part is the index, so having the `index:{}` means that it will use the index specified in the endpoint instead of a specific one.